iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/list.inl

```diff
@@ -701,7 +702,7 @@ inline bool list<T, Capacity>::isInvalidIterOrDifferentLists(const const_iterato
 template <typename T, uint64_t Capacity>
 inline void list<T, Capacity>::errorMessage(const char* source, const char* msg) noexcept
 {
-    std::cerr << source << " ::: " << msg << std::endl;
+    IOX_LOG(INFO) << source << " ::: " << msg;
```

Comment:
Having a closer look at where this is used I would also downgrade this to a debug message. It is used at `emplace` and `erase` and there also inconsistently. For example in `emplace` it just returns the `end` iterator if a wrong iterator is passed to the method but logs a message and returns the `end` iterator if there is no space left. In this case it would make much  more sense to log a debug message in the first case instead of the second.

Could you please remove this method and just add a `IOX_LOG(DEBUG)` before `return end();` in `emplace` and `erase`?